.github/workflows/pkgci_sharktank_regression.yml

```diff
@@ -0,0 +1,97 @@
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: PkgCI Sharktank Regression Tests
+on:
+  workflow_call:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+  workflow_dispatch:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+
+jobs:
+  sharktank_regresion_tests:
+    name: "sharktank_regresion_tests :: ${{ matrix.name }}"
+    runs-on: ${{ matrix.runs-on }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # CPU
+          - name: cpu_task_model
+            target: target_cpu
+            gpu: none
+            runs-on: ubuntu-24.04
+          - name: cpu_llvm_task_regression
+            models-config-file: models_cpu_llvm_task.json
+            backend: cpu
+            runs-on:
+              - self-hosted # must come first
+              - persistent-cache
+              - Linux
+              - X64
+
+          # AMD GPU
+          - name: rocm_hip_model
+            target: target_hip
+            gpu: gfx1100
+            runs-on: nodai-amdgpu-w7900-x86-64
+          - name: amdgpu_rocm_mi250_gfx90a_regression
+            rocm-chip: gfx90a
+            backend: rocm
+            sku: mi250
+            runs-on: nodai-amdgpu-mi250-x86-64
+          - name: amdgpu_rocm_mi300_gfx942_regression
+            rocm-chip: gfx942
+            backend: rocm
+            sku: mi300
+            runs-on: nodai-amdgpu-mi300-x86-64
```

Comment:
FYI we may want to migrate workflows from these `nodai-amdgpu-mi300-x86-64` runners to a new cluster of runners with the `linux-mi300-gpu-1` label: https://github.com/nod-ai/shark-ai/issues/793

That should be separate from this PR though.